apps/nextjs-website/src/lib/webinars/dynamodb/codec.ts

```diff
@@ -38,25 +47,89 @@ export const makeWebinarQuestionFromDynamodbItem = (
   webinarId: input.webinarId.S,
   question: input.question.S,
   createdAt: input.createdAt.S,
-  // The fields hiddenBy and highlightedBy can be undefined, { N: true } or { S: "string" }
-  hiddenBy:
-    !input.hiddenBy || (input.hiddenBy as unknown as DynamodbAttrNullType).NULL
-      ? undefined
-      : (input.hiddenBy as unknown as DynamodbAttrSType).S,
-  highlightedBy:
-    !input.highlightedBy ||
-    (input.highlightedBy as unknown as DynamodbAttrNullType).NULL
-      ? undefined
-      : (input.highlightedBy as unknown as DynamodbAttrSType).S,
+  hiddenBy: input.hiddenBy?.S,
+  highlightedBy: input.highlightedBy?.S,
 });
 
 export const makeDynamodbItemFromWebinarQuestion = (input: WebinarQuestion) =>
   WebinarQuestionDynamodbCodec.encode({
     webinarId: { S: input.webinarId },
     createdAt: { S: input.createdAt },
     question: { S: input.question },
-    hiddenBy: input.hiddenBy ? { S: input.hiddenBy } : { NULL: true },
-    highlightedBy: input.highlightedBy
-      ? { S: input.highlightedBy }
-      : { NULL: true },
+    ...(input.hiddenBy && { hiddenBy: { S: input.hiddenBy } }),
+    ...(input.highlightedBy && { highlightedBy: { S: input.highlightedBy } }),
   });
+
+type UpdateExpressionItem<T> = {
+  readonly fieldName: string;
+  readonly expression?: UpdateExpression<T>;
+};
+// Helper function to create UpdateExpression and ExpressionAttributeValues.
+// https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html
+const makeUpdateExpression = (
+  expressionList: ReadonlyArray<UpdateExpressionItem<string>>
+): Pick<
+  UpdateItemCommandInput,
+  'UpdateExpression' | 'ExpressionAttributeValues'
+> => {
+  // The type of the accumulator of the reduce function
+  type Zero = {
+    // the set command; e.g.: set fieldName0 = :fieldName0
+    readonly set?: string;
+    // the remove command; e.g.: remove fieldName0
+    readonly remove?: string;
+    // if ExpressionAttributeValues is empty the system raises a runtime error
+    readonly ExpressionAttributeValues?: UpdateItemCommandInput['ExpressionAttributeValues'];
```

Comment:
Why this is uppercased? It is not clear if it's a type or a value, i'd use it with expressionAttributeValues and then remap it at the end of the function when the command is constructed, something like:


```suggestion
    readonly expressionAttributeValues?: UpdateItemCommandInput['ExpressionAttributeValues'];
```